### PR TITLE
{Eventgrid} Change deprecation version

### PR DESCRIPTION
**Pull request by evelyn-ys**
**Related command**

`az eventgrid domain create`

**Description**

Push back the deprecation version as `2.46.0` doesn't allow breaking change

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Review by wangzelin007 — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/eventgrid/_params.py [hunk 1]
```diff
@@ -225,7 +225,7 @@ def load_arguments(self, _):    # pylint: disable=too-many-statements
         c.argument('public_network_access', arg_type=public_network_access_type)
         c.argument('inbound_ip_rules', action=AddInboundIpRule, nargs='+')
         c.argument('sku', arg_type=sku_type)
-        c.argument('identity', arg_type=identity_type, deprecate_info=c.deprecate(expiration='2.46.0'))
+        c.argument('identity', arg_type=identity_type, deprecate_info=c.deprecate(expiration='2.49.0'))
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/eventgrid/_params.py:228]: Best to just remove expiration?

**Comment by yonzhan** [comment]
Eventgrid

File: src/azure-cli/azure/cli/command_modules/eventgrid/_params.py (hunk 1 above)
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/eventgrid/_params.py:228]:
```suggestion
        c.argument('identity', arg_type=identity_type, deprecate_info=c.deprecate()
```

**Review by zhoxing-ms — DISMISSED** [review]

**Review by wangzelin007 — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/eventgrid/_params.py [hunk 2]
```diff
@@ -225,7 +225,7 @@ def load_arguments(self, _):    # pylint: disable=too-many-statements
         c.argument('public_network_access', arg_type=public_network_access_type)
         c.argument('inbound_ip_rules', action=AddInboundIpRule, nargs='+')
         c.argument('sku', arg_type=sku_type)
-        c.argument('identity', arg_type=identity_type, deprecate_info=c.deprecate(expiration='2.46.0'))
+        c.argument('identity', arg_type=identity_type, deprecate_info=c.deprecate()
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/eventgrid/_params.py:228]: Miss a ")"?

Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/eventgrid/_params.py:228]:
```suggestion
        c.argument('identity', arg_type=identity_type, deprecate_info=c.deprecate())
```

**Review by necusjz — APPROVED** [review]

**Review by jsntcy — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**